### ISSUE TITLE
OOM error after running some batches

### ISSUE THREAD

**Issue by yangshao** [ematvey/tensorflow-seq2seq-tutorials#21]
Hi, I tried to use the raw_rnn function. but after first few successful training, the tensor flow will throw an OOM Error:
which was originally created as op 'Train/HRED/my_model/tower_0/loop_utterances/rnn_3/while/lstm_cell/lstm_cell/MatMul', defined at:
  File "train_multi_gpu.py", line 24, in <module>
    train_op,train_global_step = multi_gpu_model(Model, is_training=True, scope=train_scope)
  File "/mnt/home/yangshao/social_bot/hred/dl_utils.py", line 186, in multi_gpu_model
    model = model_class(is_training, scope)
  File "/mnt/home/yangshao/social_bot/hred/hred.py", line 28, in __init__
    self.build_graph(emb)
  File "/mnt/home/yangshao/social_bot/hred/hred.py", line 172, in build_graph
    decoder_outpuots_ta, decoder_final_state, _ = tf.nn.raw_rnn(self.decoder_cell, self.loop_fn)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/python/ops/rnn.py", line 1043, in raw_rnn
    swap_memory=swap_memory)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2623, in while_loop
    result = context.BuildLoop(cond, body, loop_vars, shape_invariants)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2456, in BuildLoop
    pred, body, original_loop_vars, loop_vars, shape_invariants)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2406, in _BuildLoop
    body_result = body(*packed_vars_for_body)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/python/ops/rnn.py", line 987, in body
    (next_output, cell_state) = cell(current_input, state)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 404, in __call__
    lstm_matrix = _linear([inputs, m_prev], 4 * self._num_units, bias=True)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 1048, in _linear
    res = math_ops.matmul(array_ops.concat(args, 1), weights)
  File "/mnt/home/yangshao/tf1.1/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py", line 1801, in matmul
    a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)

ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[812,2048]
	 [[Node: Train/HRED/my_model/tower_0/gradients/Train/HRED/my_model/tower_0/loop_utterances/rnn_3/while/lstm_cell/lstm_cell/MatMul_grad/MatMul_1 = MatMul[T=DT_FLOAT, transpose_a=true, transpose_b=false, _device="/job:localhost/replica:0/task:0/gpu:0"](Train/HRED/my_model/tower_0/gradients/Train/HRED/my_model/tower_0/loop_utterances/rnn_3/while/lstm_cell/lstm_cell/MatMul_grad/MatMul_1/StackPop, Train/HRED/my_model/tower_0/gradients/Train/HRED/my_model/tower_0/loop_utterances/rnn_3/while/lstm_cell/lstm_cell/BiasAdd_grad/tuple/control_dependency)]]
	 [[Node: Train/HRED/Adam/update/NoOp_1/_52370 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_7406_Train/HRED/Adam/update/NoOp_1", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

The tensor flow version I use is 1.1.0? Is there anyone have the same problem?

**Comment by ematvey** [comment]
Are you submitting this issue to the correct repo? I don't recognize any code from your stacktrace.

**Comment by ematvey** [comment]
In general GPU OOM easily occurs when you are training too large of a model or using too large batch size for your GPU.

**Comment by yangshao** [comment]
thanks， i figured it out！
